java/src/main/java/io/cucumber/cucumberexpressions/RegexpUtils.java

```diff
@@ -0,0 +1,70 @@
+package io.cucumber.cucumberexpressions;
+
+class RegexpUtils {
+    /**
+     * List of characters to be escaped.
+     * The last char is '}' with index 125, so we need only 126 characters.
+     */
+    private static final boolean[] CHAR_TO_ESCAPE = new boolean[126];
+    static {
+        CHAR_TO_ESCAPE['^'] = true;
+        CHAR_TO_ESCAPE['$'] = true;
+        CHAR_TO_ESCAPE['['] = true;
+        CHAR_TO_ESCAPE[']'] = true;
+        CHAR_TO_ESCAPE['('] = true;
+        CHAR_TO_ESCAPE[')'] = true;
+        CHAR_TO_ESCAPE['{'] = true;
+        CHAR_TO_ESCAPE['}'] = true;
+        CHAR_TO_ESCAPE['.'] = true;
+        CHAR_TO_ESCAPE['|'] = true;
+        CHAR_TO_ESCAPE['?'] = true;
+        CHAR_TO_ESCAPE['*'] = true;
+        CHAR_TO_ESCAPE['+'] = true;
+        CHAR_TO_ESCAPE['\\'] = true;
+    }
+
+    /**
+     * Escapes the regexp characters (the ones from "^$(){}[].+*?\")
+     * from the given text, so that they are not considered as regexp
+     * characters.
+     * @param text the non-null input text
+     * @return the input text with escaped regexp characters
+     */
+    public static String escapeRegex(String text) {
+        /*
+        Note on performance: this code has been benchmarked for
+        escaping frequencies of 100%, 50%, 20%, 10%, 1%, 0.1%.
+        Amongst 4 other variants (including Pattern matching),
+        this variant is the faster on all escaping frequencies.
+        */
+        int length = text.length();
+        StringBuilder sb = null; // lazy initialization
+        int blocStart=0;

```

Comment:
Perhaps `blockStart` would be better..

And the formatting is a bit inconsistent, here and a few other places. If possible, please use a formatter.